### ISSUE TITLE
Feature: Allow choosing compression levels for formats that support them

### ISSUE THREAD

**Issue by EpicLPer** [C4illin/ConvertX#39]
Heya,

I'm trying to convert a few files into FLAC, which supports different compression levels. It would be nice if the GUI had an option to choose different levels, as of right now it just uses level 0 which generates rather huge files (compared to OGG).

A bit offtopic to this issue but it might in general be nice to edit the ffmpeg options (maybe as an "Advanced" feature to be enabled via docker variables globally?) or have pre-defined presets you can create :) Right now ffmpeg itself decides what options it'll use by default, which sometimes uses too high of a quality for whatever input file you give it.

Thanks!

**Comment by C4illin** [comment]
Yes I was thinking of having some options per converter but haven't found the time yet :)

But the default values should be good, for ease of use. Is 0 a good default?

**Comment by EpicLPer** [comment]
Compression is lossless in this case, it just affects the filesize in the end. Converting a ~8MB OGG to a 50MB FLAC (compression 0) is a bit "too much" in my eyes tho ^^"
With higher compressions these go down significantly.

**Comment by IkariDevGIT** [comment]
> Yes I was thinking of having some options per converter but haven't found the time yet :)

Could we please get this? @C4illin

**Comment by big-pdf** [comment]
> Yes I was thinking of having some options per converter but haven't found the time yet :)
> 
> But the default values should be good, for ease of use. Is 0 a good default?

I would also like this! Would be nice to have the possibility when for instance converting images from heic -> jpg to also downgrade the quality of the target image at the same time. I figure it could be a small expandable "Advanced" pane that you click to expand where you can set some additional conversion options as image size or quality.

**Comment by mokahless** [comment]
I think this request should be the main goal so I'm not going to open another about my issues specifically unless the author says they don't plan on doing this anymore.  Currently, jpeg with ffmpeg is set to absolute trash quality levels. Meanwhile, all the other 3 (imagemagick, vips and graphicsmagick) are set to reasonable levels, albeit too high quality for my web needs.

If we can't get configurable in web, can we at least get configurable with environment variables or a config file?
